utils/S3UtilsMongoClient.js

```diff
@@ -207,153 +214,197 @@ class S3UtilsMongoClient extends MongoClientInterface {
-                            });
-                        }
-                    });
-                    Object.keys(data.account).forEach(account => {
-                        if (!collRes.account[account].locations) {
-                            collRes.account[account].locations = {};
-                        }
+                // Dynamically get the metrics based on the object state
+                if (isObjectCold) {
+                    targetCount += 'Cold';
+                    targetData += 'Cold';
+                } else if (isObjectRestoring) {
+                    targetCount += 'Restoring';
+                    targetData += 'Restoring';
+                } else if (isObjectRestored) {
+                    targetCount += 'Restored';
+                    targetData += 'Restored';
+                }
+
+                if (isMPUPart || isOverviewKey) {
+                    targetCount = 'mpuUploadCounts';
+                    targetData = 'mpuPartsData';
+                }
 
-                        Object.keys(data.location).forEach(location => {
-                            if (!collRes.account[account].locations[location]) {
-                                collRes.account[account].locations[location] = {
+                Object.keys(data).forEach(metricLevel => {
+                    // metricLevel can only be 'bucket', 'location' or 'account'
+                    if (validStorageMetricLevels.has(metricLevel)) {
+                        Object.keys(data[metricLevel]).forEach(resourceName => {
+                            // resourceName can be the name of bucket, location or account
+                            if (!collRes[metricLevel][resourceName]) {
+                                collRes[metricLevel][resourceName] = {
                                     ...baseMetricsObject,
                                 };
                             }
-                            collRes.account[account].locations[location][targetData] += data.location[location];
-                            collRes.account[account].locations[location][targetCount]++;
-                            collRes.account[account].locations[location].deleteMarkerCount += res.value.isDeleteMarker ? 1 : 0;
+                            collRes[metricLevel][resourceName][targetData] += data[metricLevel][resourceName];
+                            // Do not count the MPU parts as objects
+                            if (!isMPUPart) {
+                                collRes[metricLevel][resourceName][targetCount]++;
+                            }
+                            collRes[metricLevel][resourceName].deleteMarkerCount += entry.value.isDeleteMarker ? 1 : 0;
                         });
+                    }
+                });
+                Object.keys(data.account).forEach(account => {
+                    if (!collRes.account[account].locations) {
+                        collRes.account[account].locations = {};
+                    }
+
```

Comment:
nit: introduce variable
```suggestion
const locations = collRes.account[account].locations;
```